src/ape_cache/query.py

```diff
@@ -0,0 +1,312 @@
+from pathlib import Path
+from typing import Any, Dict, Iterator, List, Optional
+
+import pandas as pd
+from sqlalchemy import create_engine
+from sqlalchemy.engine import CursorResult  # type: ignore
+from sqlalchemy.sql import insert, text
+from sqlalchemy.sql.expression import Insert, TextClause
+
+from ape.api import QueryAPI, QueryType
+from ape.api.networks import LOCAL_NETWORK_NAME
+from ape.api.query import BaseInterfaceModel, BlockQuery, BlockTransactionQuery, ContractEventQuery
+from ape.exceptions import QueryEngineError
+from ape.logging import logger
+from ape.utils import singledispatchmethod  # type: ignore
+
+from . import models
+from .models import Blocks, ContractEvents, Transactions
+
+
+class CacheQueryProvider(QueryAPI):
+    """
+    Default implementation of the ape.api.query.QueryAPI
+    Allows for the query of blockchain data using connected provider
+    """
+
+    # Database management
+    def get_database_file(self, ecosystem_name: str, network_name: str) -> Path:
+        """
+        Let's us figured out what the file *will be*, mostly used for database management
```

Comment:
```suggestion
        Allows us to figure out what the file *will be*, mostly used for database management.
```

I think `lets` without the `'` is the correct on here